flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/PartitionRequestClientFactoryTest.java

```diff
@@ -49,6 +51,18 @@
 
 	private final static int SERVER_PORT = NetUtils.getAvailablePort();
 
+	// see https://issues.apache.org/jira/browse/FLINK-18821
+	@Test(expected = IOException.class)
+	public void testFailureReportedToSubsequentRequests() throws Exception {
+		PartitionRequestClientFactory factory = new PartitionRequestClientFactory(new FailingNettyClient());
+		try {
+			factory.createPartitionRequestClient(new ConnectionID(new InetSocketAddress(InetAddress.getLocalHost(), 8080), 0));
+		} catch (IOException e) {
+			// expected
+		}
+		factory.createPartitionRequestClient(new ConnectionID(new InetSocketAddress(InetAddress.getLocalHost(), 8080), 0));
```

Comment:
nit: define the local `connectionId` for above two places. 
It is up to you whether to adjust it. I am also fine with current way.

Reply:
You're right, but I think it's not very important in tests and 1.11.